packages/blade/src/utils/useBreakpoint/useBreakpoint.ts

```diff
@@ -0,0 +1,144 @@
+                mediaQuery = `screen and (min-width: ${
+                  breakpointsArray[index - 1][1] + 1
+                }px) and (max-width: ${breakpointSize}px)`;
+              } else {
+                mediaQuery = `screen and (max-width: ${breakpointSize}px)`;
+              }
+
+              return [breakpointTokenName, breakpointSize, mediaQuery];
+            },
+          )
+        : []) as [keyof Breakpoints, number, string][],
+    [breakpoints, supportsMatchMedia],
+  );
+
+  const getMatchedDeviceType = useCallback((matchedBreakpoint: Breakpoint): DeviceType => {
+    let matchedDeviceType: DeviceType = 'mobile';
+    if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
+      // react-native
+      matchedDeviceType = 'mobile';
+    } else if (typeof document !== 'undefined') {
+      // browser
+      if (matchedBreakpoint && ['xs', 's', 'm'].includes(matchedBreakpoint)) {
+        // tablet is also categorised as mobile
+        matchedDeviceType = 'mobile';
+      } else {
+        matchedDeviceType = 'desktop';
+      }
+    } else if (typeof process !== 'undefined') {
+      // node
+      //@TODO: Check for useragent for node
+      matchedDeviceType = 'desktop';
+    }
+    return matchedDeviceType;
+  }, []);
+
+  const getMatchedBreakpoint = useCallback(
+    (event?: MediaQueryListEvent): Breakpoint => {
+      const matchedBreakpoint =
+        breakpointsTokenAndQueryCollection.find(([_, __, mediaQuery]) => {
+          if (event?.media === mediaQuery) {
+            return true;
+          }
+          if (window.matchMedia(mediaQuery).matches) {
+            return true;
+          }
+          return false;
+        })?.[0] ?? undefined;
+
+      return matchedBreakpoint;
+    },
+    [breakpointsTokenAndQueryCollection],
+  );
+
+  const [breakpointAndDevice, setBreakpointAndDevice] = useState(() => {
+    const matchedBreakpoint = getMatchedBreakpoint();
+    const matchedDeviceType = getMatchedDeviceType(matchedBreakpoint);
+    return {
+      matchedBreakpoint,
+      matchedDeviceType,
+    };
+  });
+
+  useEffect(() => {
+    if (!supportsMatchMedia) {
+      return undefined;
+    }
+
+    const handleMediaQueryChange = (event: MediaQueryListEvent): void => {
+      setBreakpointAndDevice((prevBreakpointAndDevice) => {
+        const matchedBreakpoint = getMatchedBreakpoint(event);
+        const matchedDeviceType = getMatchedDeviceType(matchedBreakpoint);
+        if (
+          prevBreakpointAndDevice.matchedBreakpoint !== matchedBreakpoint ||
+          prevBreakpointAndDevice.matchedDeviceType !== matchedDeviceType
+        ) {
+          return { matchedBreakpoint, matchedDeviceType };
+        }
+        return prevBreakpointAndDevice;
+      });
+    };
+
+    const mediaQueryInstances = breakpointsTokenAndQueryCollection.map(([_, __, mediaQuery]) =>
+      window.matchMedia(mediaQuery),
+    );
+
+    mediaQueryInstances.forEach((mediaQueryInstance) => {
```

Comment:
now if you see we basically iterate on all the instances we got above in `mediaQueryInstances` and attach a `change` event and a handler to it.

Reply:
We probably don't need this second `forEach`?? We can attach the event listeners while mapping over the mediaQueries

```ts
    const mediaQueryInstances = breakpointsTokenAndQueryCollection.map(([_, __, mediaQuery]) =>{
      const mediaQueryInstance = window.matchMedia(mediaQuery);
    
      if (mediaQueryInstance.addEventListener) {
        mediaQueryInstance.addEventListener('change', handleMediaQueryChange);
      } else {
        // In older browsers MediaQueryList do not yet inherit from EventTarget, So using addListener as fallback - https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener
        mediaQueryInstance.addListener(handleMediaQueryChange);
      }

      return mediaQueryInstance;
    });
```

Reply:
+1

Can even do `.map.forEach`.

Reply:
`.map.forEach` would be the same as before, 
What I'm saying is to basically do that inside map itself. That way we won't need to loop twice. once in `map` and then on `forEach`